pkg/certificate/manager.go

```diff
@@ -106,16 +105,34 @@ func (m *Manager) getFromCache(cn CommonName) *Certificate {
 
 // IssueCertificate implements Manager and returns a newly issued certificate from the given client.
 func (m *Manager) IssueCertificate(cn CommonName, validityPeriod time.Duration) (*Certificate, error) {
+	// var additionalRoot pem.RootCertificate
+	var err error
+	cert := m.getFromCache(cn) // Don't call this while holding the lock
+
+	m.mu.RLock()
+	pubIssuer := m.pubIssuer
+	keyIssuer := m.keyIssuer
+	m.mu.RUnlock()
+
 	start := time.Now()
+	if cert == nil || cert.keyIssuer != keyIssuer.ID || cert.pubIssuer != pubIssuer.ID {
+		fmt.Println("issuing!")
+		cert, err = keyIssuer.IssueCertificate(cn, validityPeriod)
+		if err != nil {
+			return nil, err
+		}
+		if pubIssuer.ID != keyIssuer.ID {
+			pubCert, err := pubIssuer.IssueCertificate(cn, validityPeriod)
+			if err != nil {
+				return nil, err
+			}
 
-	if cert := m.getFromCache(cn); cert != nil {
-		return cert, nil
-	}
+			cert = cert.newMergedWithRoot(pubCert.GetIssuingCA())
+		}
 
-	// TODO(#4502): determine client(s) to use based on the rotation stage(s) of the client(s)
-	cert, err := m.clients[0].IssueCertificate(cn, validityPeriod)
-	if err != nil {
-		return cert, err
+		cert.keyIssuer = keyIssuer.ID
+		cert.pubIssuer = m.pubIssuer.ID
```

Comment:
Why is `m.pubIssuer.ID` vs `pubIssuer.ID`

Reply:
good catch! fixed